pkg/cluster/util.go

```diff
@@ -483,3 +495,7 @@ func (c *Cluster) GetSpec() (*acidv1.Postgresql, error) {
 func (c *Cluster) patroniUsesKubernetes() bool {
 	return c.OpConfig.EtcdHost == ""
 }
+
+func (c *Cluster) needConnectionPool() bool {
+	return c.Spec.ConnectionPool != nil || c.Spec.EnableConnectionPool == true
```

Comment:
So, when I define a `connectionPool` field, but set `enableConnectionPool` to `false` it will still get created?

Reply:
At the moment yes. The idea is that both ways are independent, but I see
what you mean - in the case if someone has both `enableConnectionPool`
and `connectionPool` and suddenly decide to turn it off (maybe
temporary), but keep configuration, it could be confusing. Maybe we can
make `enableConnectionPool` of `*bool`, and check `connectionPool` only
if `enableConnectionPool` is not specified, i.e. nil.